src/premint/ZoraCreator1155PremintExecutor.sol

```diff
@@ -162,4 +168,27 @@ contract ZoraCreator1155PremintExecutor {
             isValid = IZoraCreator1155(contractAddress).isAdminOrRole(recoveredSigner, CONTRACT_BASE_ID, PERMISSION_BIT_MINTER);
         }
     }
+
+    // upgrade related functionality
+
+    /// @notice The name of the contract for upgrade purposes
+    function contractName() external pure returns (string memory) {
```

Comment:
thoughts on having this be `contractVersion` instead ? @oveddan @iainnash

Reply:
we can also add a contract version - this is mainly used when upgrading, to make sure the contract upgrading to has the same `contractName` as this one, that wouldn't quite make sense in the context of versions, but there could be another check that ensures the version is increasing.

Reply:
i think actually adding the contract version makes sense too, just so we have it on record

Reply:
ah i see yeah lets keep